### ISSUE TITLE
out of bound player data

### ISSUE THREAD

**Issue by mellotanica** [Music-Machine-Learning/IMprovEEsation#10]
player is the dispatcher of external events to internal components. it should be able to process incoming messages and operate consequently. external events come from a rpi pedalboard right now through tcp connection. external signals can be midi events to be injected into midi flow or virtual directives to be forwarded to internal components

**Comment by matteomartelli** [comment]
May the director need to handle MIDI commands from the pedalboard as well? Thus, wouldn't be better the use of a new identità/process as a dispatcher of the user interaction events? The "interactor"!!

**Comment by mellotanica** [comment]
player will trigger director measure generation in "live synchronous mode", furthermore most out of bound interaction will be composed of messages aimed to the player itself. it's much more optimizable to implement everything inside the player

**Comment by matteomartelli** [comment]
Fair enough but...did you consider the stress cost needed in a player code handling? O_O

**Comment by berdav** [comment]
This can be done modifying the network API and passing this information as out of BAND data placing a callback function in the player code. Hopefully this modification will leave the player almost untouched (except the recv_to_play function)
